pkg/fleet-manager/migrate_controller.go

```diff
@@ -0,0 +1,292 @@
+	fleetClusters, err := fetchDestinationClusters(ctx, m.Client, migrate.Namespace, migrate.Spec.SourceCluster)
+	if err != nil {
+		log.Error(err, "Failed to fetch source cluster for migration")
+		return ctrl.Result{}, fmt.Errorf("fetching source cluster: %w", err)
+	}
+	var sourceClusterKey ClusterKey
+	var sourceClusterAccess *fleetCluster
+	// "migrate.Spec.SourceCluster" must contain one clusters, it is ensured by admission webhook
+	for key, value := range fleetClusters {
+		sourceClusterKey = key
+		sourceClusterAccess = value
+	}
+
+	// Construct Velero backup instance
+	migrateLabel := generateVeleroInstanceLabel(MigrateNameLabel, migrate.Name, migrate.Spec.SourceCluster.Fleet)
+	sourceBackupName := generateVeleroResourceName(sourceClusterKey.Name, MigrateKind, migrate.Namespace, migrate.Name)
+	sourceBackup := buildVeleroBackupFromMigrate(&migrate.Spec, migrateLabel, sourceBackupName)
+
+	// Sync the Velero backup resource
+	if err = syncVeleroObj(ctx, sourceClusterAccess, sourceBackup); err != nil {
+		log.Error(err, "Failed to create backup resource for migration", "backupName", sourceBackupName)
+		return ctrl.Result{}, fmt.Errorf("creating backup resource: %w", err)
+	}
+
+	// Get the status of Velero backup resources
+	veleroBackup := &velerov1.Backup{}
+	err = getResourceFromClusterClient(ctx, sourceBackupName, VeleroNamespace, *sourceClusterAccess, veleroBackup)
+	if err != nil {
+		log.Error(err, "Failed to retrieve backup resource for migration", "backupName", sourceBackupName)
+		return ctrl.Result{}, fmt.Errorf("retrieving backup status: %w", err)
+	}
+
+	// Update migration status based on the Velero backup details
+	currentBackupDetails := &backupapi.BackupDetails{
+		ClusterName:           sourceClusterKey.Name,
+		ClusterKind:           sourceClusterKey.Kind,
+		BackupNameInCluster:   veleroBackup.Name,
+		BackupStatusInCluster: &veleroBackup.Status,
+	}
+	migrate.Status.SourceClusterStatus = currentBackupDetails
+
+	if veleroBackup.Status.Phase == velerov1.BackupPhaseCompleted {
+		conditions.MarkTrue(migrate, backupapi.SourceReadyCondition)
+	}
+
+	return ctrl.Result{}, nil
+}
+
+// reconcileMigrateRestore handles the restore stage of the migration process.
+func (m *MigrateManager) reconcileMigrateRestore(ctx context.Context, migrate *backupapi.Migrate) (ctrl.Result, error) {
+	log := ctrl.LoggerFrom(ctx)
+
+	// If source cluster's backup resource is not ready, return directly.
+	if !isMigrateSourceReady(migrate) {
+		return ctrl.Result{RequeueAfter: RequeueAfter}, nil
+	}
+
+	targetClusters, err := fetchDestinationClusters(ctx, m.Client, migrate.Namespace, migrate.Spec.TargetClusters)
+	if err != nil {
+		log.Error(err, "Failed to fetch target clusters for migration")
+		return ctrl.Result{}, fmt.Errorf("fetching target clusters: %w", err)
+	}
+
+	if migrate.Status.Phase != backupapi.MigratePhaseInProgress {
+		migrate.Status.Phase = backupapi.MigratePhaseInProgress
```

Comment:
I think this is confusing,  InProgress should be the interval between backup begin and restore complete.